app/src/main/java/io/reark/rxgithubapp/advanced/data/stores/cores/GsonStoreCoreBase.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2013-2016 reark project contributors
+ *
+ * https://github.com/reark/reark/graphs/contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package io.reark.rxgithubapp.advanced.data.stores.cores;
+
+import android.content.ContentResolver;
+import android.support.annotation.NonNull;
+
+import com.google.gson.Gson;
+
+import io.reark.reark.data.stores.cores.ContentProviderStoreCore;
+import io.reark.reark.utils.Preconditions;
+
+public abstract class GsonStoreCoreBase<T, U> extends ContentProviderStoreCore<T, U> {
+
+    private final Gson gson;
+
+    public GsonStoreCoreBase(@NonNull ContentResolver contentResolver, @NonNull Gson gson) {
+        super(contentResolver);
+
+        Preconditions.checkNotNull(gson, "Gson cannot be null.");
```

Comment:
Is there a special reason why would you use it like this: `this.gson = Preconditions.checkNotNull(gson, "Gson cannot be null.");`?